up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,615 @@ SPDX-FileType: DOCUMENTATION
+* contains a _wildcard_ authority or
+* contains a _wildcard_ uEntity ID (`ue_id`) or
+* contains a _wildcard_ resource ID,
 
-The uSubscription service also allows to indicate to subscribers how they shall consume published events by means of the `EventDeliveryConfig` returned in the `SubscriptionResponse` message. For example, if the subscriber has to consume from a different topic or from different messaging infrastructure, this message will store the corresponding delivery semantics.
+a uSubscription service *MUST* return a failure status message with link:../../../up-core-api/uprotocol/v1/ucode.proto[`UCode`] `INVALID_ARGUMENT`.
+--
 
-NOTE: Delivery semantics (if any) are deployment specific and not covered in this specification
+[#unregister-for-notifications-operation]
+==== UnregisterForNotifications()
+
+[.specitem,oft-sid="req~usubscription-unregister-notifications-protobuf~1",oft-needs="impl"]
+--
+The uSubscription service `UnregisterForNotifications()` function *MUST* implement the corresponding protobuf definition in link:../up-core-api/uprotocol/core/uSubscription/v3/usubscription.proto[usubscription.proto].
+--
+
+[.specitem,oft-sid="dsn~usubscription-unregister-notifications~1",oft-needs="impl,utest"]
+--
+After they have unregistered to receive such custom notifications via `UnregisterForNotifications()`, uSubscription service *MUST* stop sending subscription change notification messages to clients on a custom topic of their choice.
+--
+
+[.specitem,oft-sid="dsn~usubscription-unregister-notifications-invalid-topic~1",oft-needs="impl,utest"]
+--
+When receiving a `UnregisterForNotifications()` request that contains a topic that
+* is not a valid xref:../../../basics/uri.adoc[uProtocol URI] or
+* contains a _wildcard_ authority or
+* contains a _wildcard_ uEntity ID (`ue_id`) or
+* contains a _wildcard_ resource ID,
+
+a uSubscription service *MUST* return a failure status message with link:../../../up-core-api/uprotocol/v1/ucode.proto[`UCode`] `INVALID_ARGUMENT`.
+--
+
+== Timeout & Retry Logic
+
+Subscribe (and unsubscribe) to remote topics are handled by RPC calls between uSubscription services running on the different devices. Given that devices are not always connected to each other, the onus is on uSubscription service to ensure that a command is received in time. Below are the common retry and timeout policies for USubscription service implementations to follow:   
+
+[.specitem,oft-sid="req~usubscription-remote-max-timeout~1",oft-needs="impl"]
+--
+Remote Subscribe/Unsubscribe requests *MUST* implement a minimum timeout of 5 minutes.
+--
+
+[.specitem,oft-sid="req~usubscription-remote-retry-indefinitely~1",oft-needs="impl"]
+--
+Timed-out remote commands *MUST* be retried indefinitely until the business logic behind it no longer requires the command to be sent (e.g. because the last entity unsubscribed from a topic that is in state `SUBSCRIPTION_PENDING`).
```

Comment:
```suggestion

- Timed-out remote commands *MUST* be retried indefinitely until the business logic behind it no longer requires the command to be sent (e.g. because the last entity unsubscribed from a topic that is in state `SUBSCRIPTION_PENDING`).

```